### PR TITLE
OADP-205 Fetch and decode secret for registry plugin environment

### PR DESCRIPTION
**Pull request by kaovilai**
This PR fetches secret and decode them for use in Registry "environment variables" parameters for initializing plugin registry.

Previous to this, registry environment variables returned from provider functions were wrongly assumed to be set in envVar.Value exclusively. This PR add support for envVar.ValueFrom.SecretKeyRef.
[OADP-205](https://issues.redhat.com//browse/OADP-205)

### CODE REVIEW

File: velero-plugins/imagestream/shared.go [hunk 1]
```diff
@@ -92,16 +95,42 @@ func GetRegistryEnvsForLocation(location string, namespace string) ([]string, er
...
 	for _, envVar := range envVars {
-		envVarsStr = append(envVarsStr, coreV1EnvVarToString(envVar))
+		envVarsStr = append(envVarsStr, coreV1EnvVarToString(envVar, namespace))
 	}
 	return envVarsStr
 }
-func coreV1EnvVarToString(envVar corev1.EnvVar) string {
+func coreV1EnvVarToString(envVar corev1.EnvVar, namespace string) string {
+	if envVar.ValueFrom != nil {
+		if envVar.ValueFrom.SecretKeyRef != nil {
```
Comment (dymurray) [inline velero-plugins/imagestream/shared.go]: Could combine this with Line 109

**Review by dymurray — APPROVED** [review]
changes look sane to me

File: velero-plugins/imagestream/shared.go (hunk 1 above)
Comment (kaovilai) [reply velero-plugins/imagestream/shared.go]: I was thinking maybe in the future secretkeyref will be one of many.. valueFrom used.. but rethinking now that's probably not gonna happen.. more likely will be refactoring to not create corev1.Envvar in the first place.
Will fix this.

Comment (kaovilai) [reply velero-plugins/imagestream/shared.go]: resolved in c3f4d94...14f229e

**Review by shubham-pampattiwar — APPROVED** [review]

**Comment by kaovilai** [comment]
local build succeeded skipping PR travis. Will wait for master travis.

**Pull request merged**